courier/template/load_template.go

```diff
@@ -61,12 +71,39 @@ func loadBuiltInTemplate(filesytem fs.FS, name string, html bool) (Template, err
 		tpl = t
 	}
 
-	_ = cache.Add(name, tpl)
+	_ = Cache.Add(name, tpl)
 	return tpl, nil
 }
 
+func loadRemoteTemplate(ctx context.Context, d templateDependencies, url string, name string, html bool, root string) (Template, error) {
+	if t, found := Cache.Get(name); found {
```

Comment:
The `name` is a fixed string, it is always `recovery/invalid/email.body.plaintext.gotmpl` for example. If you update your configuration, or if you have a system where you have multiple templates for the same name, this cache will break that as the cache will always return the first template it found for `recovery/invalid/email.body.plaintext.gotmpl`.

Please first add a test verifying that this is an issue (changing the template remote source does in fact not have an effect), and then remove this cache.